### PR TITLE
Db interface

### PR DESCRIPTION
**Pull request by TimothyStiles**
Many people have suggested their DB backend of choice.

TBH I don't want to mess with SQL. Some people have interest in neo4j. I don't like any solutions right now so until I can find something that personally satisfies me I've created a basic interface that individuals can use to hook up there DB backend of choice to allbase.

There still is a lot to do interface design wise. Still finalizing structs, method signatures, etc and I will be actively working on this.

### CODE REVIEW

**Review by Koeng101 — CHANGES_REQUESTED** [review]
> a basic interface that individuals can use to hook up there DB backend of choice to allbase.

IMO this seems like the pathway to a hot mess of garbage. It also seems like you're thinking of databases as "key value store", which is not the way to think about relational databases. The structure of the data matters a lot for performance and maintenance overhead. 

I think it is much better for the project if you go figure out which database satisfies you, rather than trying to slap on a poorly thought out interface that we'll have to be tied to.

File: db/db.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	// Get(key string) (string, error)
+
+	// // Set sets the value for the given key.
+	// Set(key, value string) error
+
+	// Delete removes the value for the given key.
+	Delete(key string) error
+
+	// Query returns a list of key/value pairs for the given query.
+	Query(query string) ([]map[string]string, error)
```
[the remark this answers is not in the document]
Comment (TimothyStiles) [reply db/db.go:28]: Yeah this is just a place holder generated by github co-pilot. Got a better suggestion? Could use generics or just a dedicated struct.

File: db/db.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	SetConfig(Config) error
+
+	// // Get returns the value for the given key.
+	// Get(key string) (string, error)
+
+	// // Set sets the value for the given key.
+	// Set(key, value string) error
+
+	// Delete removes the value for the given key.
+	Delete(key string) error
```
[the remark this answers is not in the document]
Comment (TimothyStiles) [reply db/db.go:25]: auto-generated co-pilot place holder. Would probably just delete on a unique ID. Any other ideas?

**Comment by TimothyStiles** [comment]
stale and no longer necessary. Closing.

**Pull request closed without merging**